### PR TITLE
LPS-187219

### PR DESCRIPTION
**Pull request by icgsbr**
## Related LPS

- [LPS-187219](https://issues.liferay.com/browse/LPS-187219)

### CODE REVIEW

**Comment by icgsbr** [comment]
ci:test:relevant

**Comment by icgsbr** [comment]
ci:test:stable

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/java/com/liferay/dynamic/data/mapping/form/field/type/internal/document/library/DocumentLibraryDDMFormFieldTemplateContextContributor.java [hunk 1]
```diff
@@ -382,6 +384,31 @@ private Map<String, Object> _getFileEntryParameters(
...
+						UriBuilder uriBuilder = UriBuilder.fromUri(
+							ddmFormFieldFileEntryURL);
+
+						uriBuilder.replaceQueryParam(
+							"_com_liferay_dynamic_data_mapping_form_web" +
+								"_portlet_DDMFormPortlet_fileEntryId",
+							fileEntry.getFileEntryId());
+
+						return uriBuilder.build(
+						).toString();
```
Comment (carolmariaabb) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/java/com/liferay/dynamic/data/mapping/form/field/type/internal/document/library/DocumentLibraryDDMFormFieldTemplateContextContributor.java:408]: @icgsbr please simplify this code to:

```
String portletNamespace = _portal.getPortletNamespace(
	DDMPortletKeys.DYNAMIC_DATA_MAPPING_FORM);

return UriBuilder.fromUri(
	fileEntryURL
).replaceQueryParam(
	portletNamespace + "fileEntryId",
	fileEntry.getFileEntryId()
).build(
).toString();
```

**Review by carolmariaabb — CHANGES_REQUESTED** [review]

**Comment by icgsbr** [comment]
Thanks for the comments @carolmariaabb! I'm on it!

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/java/com/liferay/dynamic/data/mapping/form/field/type/internal/document/library/DocumentLibraryDDMFormFieldTemplateContextContributor.java [hunk 2]
```diff
@@ -382,6 +384,31 @@ private Map<String, Object> _getFileEntryParameters(
 					RequestBackedPortletURLFactoryUtil.create(
 						ddmFormFieldRenderingContext.getHttpServletRequest());
 
+				if (ddmFormField.hasProperty("fileEntryURL")) {
+					long ddmFormInstanceRecordId = GetterUtil.getLong(
+						ddmFormFieldRenderingContext.getProperty(
+							"ddmFormInstanceRecordId"));
+
+					String ddmFormFieldFileEntryURL = ddmFormField.getProperty(
```
[the remark this answers is not in the document]
Comment (icgsbr) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/java/com/liferay/dynamic/data/mapping/form/field/type/internal/document/library/DocumentLibraryDDMFormFieldTemplateContextContributor.java]: done

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/java/com/liferay/dynamic/data/mapping/form/field/type/internal/document/library/DocumentLibraryDDMFormFieldTemplateContextContributor.java [hunk 3]
```diff
@@ -382,6 +384,31 @@ private Map<String, Object> _getFileEntryParameters(
...
+						"fileEntryURL"
+					).toString();
```
[the remark this answers is not in the document]
Comment (icgsbr) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/java/com/liferay/dynamic/data/mapping/form/field/type/internal/document/library/DocumentLibraryDDMFormFieldTemplateContextContributor.java]: done

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/java/com/liferay/dynamic/data/mapping/form/field/type/internal/document/library/DocumentLibraryDDMFormFieldTemplateContextContributor.java [hunk 4]
```diff
@@ -382,6 +384,31 @@ private Map<String, Object> _getFileEntryParameters(
...
+
+					if ((ddmFormFieldFileEntryURL != null) &&
+						(ddmFormInstanceRecordId == 0)) {
```
[the remark this answers is not in the document]
Comment (icgsbr) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/java/com/liferay/dynamic/data/mapping/form/field/type/internal/document/library/DocumentLibraryDDMFormFieldTemplateContextContributor.java]: done

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/java/com/liferay/dynamic/data/mapping/form/field/type/internal/document/library/DocumentLibraryDDMFormFieldTemplateContextContributor.java (hunk 1 above)
Comment (icgsbr) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/java/com/liferay/dynamic/data/mapping/form/field/type/internal/document/library/DocumentLibraryDDMFormFieldTemplateContextContributor.java]: done

**Comment by icgsbr** [comment]
ci:test:sf

**Comment by icgsbr** [comment]
ci:test:stable

**Comment by icgsbr** [comment]
ci:test:relevant

**Comment by carolmariaabb** [comment]
@icgsbr please check https://github.com/liferay-forms/liferay-portal/pull/2241/

**Pull request closed without merging**